node/pkg/driver/device_connectivity/device_connectivity_fc.go

```diff
@@ -0,0 +1,324 @@
+/**
+ * Copyright 2019 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package device_connectivity
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"errors"
+
+	"github.com/ibm/ibm-block-csi-driver/node/logger"
+	"github.com/ibm/ibm-block-csi-driver/node/pkg/driver/executer"
+)
+
+type OsDeviceConnectivityFc struct {
+	Executer          executer.ExecuterInterface
+	Helper            OsDeviceConnectivityHelperFcInterface
+	HelperScsiGeneric OsDeviceConnectivityHelperScsiGenericInterface
+}
+
+func NewOsDeviceConnectivityFc(executer executer.ExecuterInterface) OsDeviceConnectivityInterface {
+	return &OsDeviceConnectivityFc{
+		Executer:          executer,
+		Helper:            NewOsDeviceConnectivityHelperFc(executer),
+		HelperScsiGeneric: NewOsDeviceConnectivityHelperScsiGeneric(executer),
+	}
+}
+
+func (r OsDeviceConnectivityFc) RescanDevices(lunId int, arrayIdentifiers []string) error {
+	logger.Debugf("Rescan : Start rescan on specific lun, on lun : {%v}, with array wwn : {%v}", lunId, arrayIdentifiers)
+
+	if len(arrayIdentifiers) == 0 {
+		e := &ErrorNotFoundArrayIdentifiers{lunId}
+		logger.Errorf(e.Error())
+		return e
+	}
+
+	HostIDs, err := r.Helper.GetFcHostIDs()
+	if err != nil {
+		return err
+	}
+
+	return r.HelperScsiGeneric.RescanDevices(lunId, arrayIdentifiers, HostIDs)
+}
+
+func (r OsDeviceConnectivityFc) GetMpathDevice(volumeId string, lunId int, arrayIdentifiers []string) (string, error) {
```

Comment:
Can you please clarify if there is a differance between iscsi to FC on this function?
If its only the pvi-*fc-WWN vs iscsi-IQN and all the rest is 100% the same, then please refactor this function and move it to generic ISCSI part and allow providing filepath as param so the iscsi will use iscsi-IQN and the FC will use its own. (so no need to do double UT for that and we keep the same logic)

Reply:
Yes, you're right. I will refactor it and update the code

Reply:
ok, please update the comment when u finish.

Reply:
any progress on this comment?

Reply:
Peng, can u please fix the reviewable link so it will be easier to review it?